### PR TITLE
Zishan - add batch facet

### PR DESCRIPTION
**Pull request by zishansami102**
Moving buyWithSellerFinancingBatch to a new facet

### CODE REVIEW

File: src/facets/BatchExecutionFacet.sol [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            Offer memory offer = offers[i];
+            bytes calldata signature = signatures[i];
+            uint256 tokenId = tokenIds[i];
+            uint256 tokenAmount = tokenAmounts[i];
+            address buyerRe = buyer;
+
+            // try executing current offer
+            uint256 balanceBefore = address(this).balance;
+            try
+                INiftyApes(address(this)).buyWithSellerFinancing{
```
Comment (captnseagraves) [inline src/facets/BatchExecutionFacet.sol:54]: This is the point you were making about keeping it as an external call and still being able to use try/catch? I wonder what the gas cost difference would be.

Comment (captnseagraves) [reply src/facets/BatchExecutionFacet.sol:54]: Thinking about it, having externals calls between facets seems like a bit of an anti-pattern. It does provide a work around that allows for the use of try/catch, but it feels like a misuse of the diamond pattern and external calls.

Comment (captnseagraves) [reply src/facets/BatchExecutionFacet.sol:54]: What if we just removed the single action functions and just made the batch functions the core functions. That would reduce the size and complexity of the system.

Comment (zishansami102) [reply src/facets/BatchExecutionFacet.sol:54]: It is not that uncommon in diamond patterns.
I had read an article by mudgen's regarding this, and there are alternate ways of calling functions from other facets with trade-offs.
https://eip2535diamonds.substack.com/p/how-to-share-functions-between-facets?s=w

Comment (zishansami102) [reply src/facets/BatchExecutionFacet.sol:54]: Also, if we merge the singe functions and batch functions in one, then the partial execution logic check has to be added at each and every revert points in the code, which itself will make the code a lot more bulkier and hard to read and maintain.
Abstracting the batch functionality with try-catch helps us simplify the core business logic, and allow us to implement the partial execution in an efficient manner.

Comment (zishansami102) [reply src/facets/BatchExecutionFacet.sol:54]: seize asset doesn't have partial execution logic implemented for that very reason.

Comment (zishansami102) [reply src/facets/BatchExecutionFacet.sol:54]: also, `msg.value` management needs to be configured for failures as well

**Pull request merged**